src/main/groovy/com/twcable/grabbit/jcr/JCRNodeDecorator.groovy

```diff
@@ -77,18 +83,23 @@ class JcrNodeDecorator {
     }
 
 
-    String getPrimaryType() {
-        innerNode.getProperty(JCR_PRIMARYTYPE).string
-    }
-
-
     /**
     * Identify all required child nodes
-    * @return list of immediate required child nodes that must exist with this node, or null if no children
+    * @return list of immediate required child nodes that must be transported with this node, or null if no required nodes
     */
     @Nullable
     Collection<JcrNodeDecorator> getRequiredChildNodes() {
-        return hasMandatoryChildNodes() ? getImmediateChildNodes().findAll{ JcrNodeDecorator childJcrNode -> childJcrNode.isRequiredNode() } : null
+        Collection<JcrNodeDecorator> requiredNodes
+        if(isAuthorizableType()) {
+            //Some child nodes must be transported with authorizables. This is because authorizable nodes have different UUID names from server to server
+            requiredNodes = getImmediateChildNodes().findAll { JcrNodeDecorator childJcrNode -> childJcrNode.isAuthorizablePart()}
+        }
+        else {
+            //Some nodes must be saved together, per node definition
+            requiredNodes = hasMandatoryChildNodes() ? getImmediateChildNodes().findAll{ JcrNodeDecorator childJcrNode -> childJcrNode.isRequiredNode() } : null
+        }
+
+        return requiredNodes ?: null
```

Comment:
I agree, and I actually think we can break out some of these pieces, and just have this method return the result of a ternary.